www/js/diary/timelineHelper.ts

```diff
@@ -217,3 +222,349 @@ const locations2GeojsonTrajectory = (trip, locationList, trajectoryColor?) => {
     };
   });
 };
+
+// Remaining functions from /diary/services.js
```

Comment:
I had left a comment here in the old file. Make sure to add it back (or something to that effect)

```
      // DB entries retrieved from the server have '_id', 'metadata', and 'data' fields.
      // This function returns a shallow copy of the obj, which flattens the
      // 'data' field into the top level, while also including '_id' and 'metadata.key'
```